lib/ops/opsservice/environment.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opsservice
+
+import (
+	"context"
+
+	"github.com/gravitational/gravity/lib/constants"
+	"github.com/gravitational/gravity/lib/defaults"
+	"github.com/gravitational/gravity/lib/kubernetes"
+	"github.com/gravitational/gravity/lib/ops"
+	"github.com/gravitational/gravity/lib/storage"
+
+	"github.com/gravitational/rigging"
+	"github.com/gravitational/trace"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// GetClusterEnvironment retrieves the cluster environment
+func (o *Operator) GetClusterEnvironment(key ops.SiteKey) (env storage.Environment, err error) {
+	client, err := o.GetKubeClient()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	configmap, err := client.CoreV1().ConfigMaps(defaults.KubeSystemNamespace).
```

Comment:
Why represent the object in kubernetes as a configmap instead of a crd? I think it's easier to implement as a configmap, just seems a bit odd that you get a resource implementation when using gravity, and a configmap representation when using kubectl.

Reply:
Yes, fair point. I did not think I'd bring much with turning it into a CRD other than the implementation would be more complex as the primary way of managing these resources are via the `gravity` binary.